packages/react/src/combobox/group-label/ComboboxGroupLabel.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+import * as React from 'react';
+import { useIsoLayoutEffect } from '@base-ui-components/utils/useIsoLayoutEffect';
+import { BaseUIComponentProps } from '../../utils/types';
+import { useBaseUiId } from '../../utils/useBaseUiId';
+import { useComboboxGroupContext } from '../group/ComboboxGroupContext';
+import { useRenderElement } from '../../utils/useRenderElement';
+
+/**
+ * An accessible label that is automatically associated with its parent group.
+ * Renders a `<div>` element.
+ */
+export const ComboboxGroupLabel = React.forwardRef(function ComboboxGroupLabel(
+  componentProps: ComboboxGroupLabel.Props,
+  forwardedRef: React.ForwardedRef<HTMLDivElement>,
+) {
+  const { render, className, id: idProp, ...elementProps } = componentProps;
+
+  const { setLabelId } = useComboboxGroupContext();
+
+  const id = useBaseUiId(idProp);
+
+  useIsoLayoutEffect(() => {
+    setLabelId(id);
+    return () => {
+      setLabelId(undefined);
+    };
+  }, [id, setLabelId]);
+
+  const element = useRenderElement('div', componentProps, {
+    ref: forwardedRef,
+    props: [{ id }, elementProps],
```

Comment:
Feels like we could instead easily just set `elementProps.id = ...` instead of creating a wrapper object & array.